front/lib/assistant.ts

```diff
@@ -171,6 +171,14 @@ function compareByImportance(
           .connectorProvider
     );
 
+  if (aConnector === null && bConnector != null) {
```

Comment:
This function becomes pretty hard to parse.

Can we:
1. extract the `order` compute since it does not rely on the function's parameters.
2 then we can probably do something like this:


```
function compareByImportance(
  a: ComparableByProvider,
  b: ComparableByProvider
): number {
  // Handle null cases.
  if (aConnector === null) return bConnector === null ? 0 : 1;
  if (bConnector === null) return -1;

  // Handle webcrawler cases.
  if (aConnector === webcrawlerProvider) return 1;
  if (bConnector === webcrawlerProvider) return -1;

  // Get indices in sorted connectors.
  const indexA = order.indexOf(aConnector);
  const indexB = order.indexOf(bConnector);

  // If both are not found, they are considered equal.
  if (indexA === -1 && indexB === -1) return 0;

  // Compare indices, treating not found as less important.
  return (indexA === -1 ? order.length : indexA) - (indexB === -1 ? order.length : indexB);
}
```